### ISSUE TITLE
Filters are returned in random order, so always trigger updates

### ISSUE THREAD

**Issue by passcod** [honeycombio/terraform-provider-honeycombio#233]
**Versions**

```
Terraform v1.3.2
on linux_amd64
+ provider registry.terraform.io/honeycombio/honeycombio v0.11.1
```

**Steps to reproduce**

1. Create a query with multiple filters

```terraform
data "honeycombio_query_specification" "queue_email" {
  # VISUALIZE
  calculation {
    op = "SUM"
    column = "queueCount"
  }

  # WHERE
  filter {
    column = "queueCount"
    op     = ">="
    value  = "0"
  }
  filter {
    column = "message"
    op     = "contains"
    value  = "ScheduledTask: Email"
  }
  filter {
    column = "deployment"
    op     = "="
    value  = "foobar"
  }

  # 24 hours
  time_range = 60 * 60 * 24
}

resource "honeycombio_query" "queue_email" {
  dataset    = var.dataset
  query_json = data.honeycombio_query_specification.queue_email.json
}
```

2. Apply
3. Plan again: we expect a noop, but it wants to update resources because the API returned filters in a different order

```
  # honeycombio_query.queue_email must be replaced
-/+ resource "honeycombio_query" "queue_email" {
      ~ id         = "pepJMiY21Zc" -> (known after apply)
      ~ query_json = jsonencode(
          ~ {
              ~ filters      = [
                  - {
                      - column = "deployment"
                      - op     = "="
                      - value  = "foobar"
                    },
                    {
                        column = "queueCount"
                        op     = ">="
                        value  = 0
                    },
                    {
                        column = "message"
                        op     = "contains"
                        value  = "ScheduledTask: Email"
                    },
                  + {
                      + column = "deployment"
                      + op     = "="
                      + value  = "foobar"
                    },
                ]
                # (2 unchanged elements hidden)
            } # forces replacement
        )
        # (1 unchanged attribute hidden)
    }
```

**Comment by jharley** [comment]
@passcod thank you for reporting this, and for your patience on waiting for a reply.

I've just reproduced this locally and am looking into a fix.

**Comment by jharley** [comment]
@passcod the intended fix for this just went out in the 0.11.2 release. Please let me know if you run into any issues, and thank you again for taking the time to report this issue.
